### ISSUE TITLE
Kebab typedef

### ISSUE THREAD

**Issue by priley86** [patternfly/patternfly-react#1921]
As mentioned in Slack, I am seeing the following when passing the `component` prop for the Kebab:
```
ERROR in /go/src/github.com/openshift/console/frontend/public/components/utils/kebab.tsx
(139,13): Type '"a" | "button"' is not assignable to type 'ElementType<DropdownItemProps>'.
  Type '"button"' is not assignable to type 'ElementType<DropdownItemProps>'.
```

Usage:
```
<DropdownItem key={option.label} 
            isDisabled={isDisabled} 
            onClick={option.callback}
            href={option.href}
            component={ option.href ? "a" : "button"}>
            {option.label}
          </DropdownItem>
```

Additionally, I tried to remove the `component` prop altogether and just let the `DropdownItem` component handle this for me (with the other props in tact). It rendered properly, however I noted the `onClick` behavior was different after this (and `preventDefault() ` was no longer called on my anchor items).

The suggested fix for now would just be updating the typedef (as the behavior is working fine with the `component` prop passed, it just fails the CI).

https://github.com/patternfly/patternfly-react/blob/master/packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.d.ts#L4

**Comment by priley86** [comment]
related to #1825  it seems

**Comment by priley86** [comment]
full source of the issue above is here:
https://github.com/priley86/web-ui/blob/react-tables-poc/frontend/public/components/utils/kebab.tsx#L134

**Comment by tlabaj** [comment]
Closed by PR #1924
